portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/BalloonEditor.testcase

```diff
@@ -0,0 +1,209 @@
+			BalloonEditor.toggleBalloonEditorContentStyle(optionTool = "Numbered");
+
+			AssertTextEquals(
+				locator1 =
+				"CKEditorBalloonEditorSample#PARAGRAPH_CONTENT_NUMBERED_LIST",
+				value1 = "${paragraphText}");
+
+			BalloonEditor.toggleBalloonEditorContentStyle(optionTool = "Numbered");
+
+			AssertElementPresent(
+				key_textName = "${paragraphText}",
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT");
+		}
+
+		task ("Check that text can be centered") {
+			BalloonEditor.toggleBalloonEditorContentStyle(optionTool = "Center");
+
+			AssertAttributeValue(
+				attribute1 = "style",
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT",
+				value1 = "text-align: center;");
+		}
+
+		task ("Check that text can be aligned right") {
+			BalloonEditor.toggleBalloonEditorContentStyle(optionTool = "Align Right");
+
+			AssertAttributeValue(
+				attribute1 = "style",
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT",
+				value1 = "text-align: right;");
+		}
+
+		task ("Check that text can be justified") {
+			BalloonEditor.toggleBalloonEditorContentStyle(optionTool = "Justify");
+
+			AssertAttributeValue(
+				attribute1 = "style",
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT",
+				value1 = "text-align: justify;");
+		}
+	}
+
+	@description = "Assert toolbar with hyperlink icon is present"
+	@priority = "4"
+	test CanManageHyperlink {
+		task ("Click the hyperlinked text") {
+			Click(locator1 = "CKEditorBalloonEditorSample#BODY_TEXT_HYPERLINK");
+		}
+
+		task ("Assert the presence of the hyperlink toolbar elements") {
+			AssertElementPresent(locator1 = "CKEditorBalloonEditorSample#HYPERLINK_UNLINK");
+
+			AssertElementPresent(locator1 = "CKEditorBalloonEditorSample#HYPERLINK_TARGET");
+
+			AssertElementPresent(locator1 = "CKEditorBalloonEditorSample#HYPERLINK_TEXT_INPUT");
+
+			AssertElementPresent(locator1 = "CKEditorBalloonEditorSample#HYPERLINK_BROWSE_SRVR_BTN");
+
+			AssertElementPresent(locator1 = "CKEditorBalloonEditorSample#CHK_BTN");
+		}
+
+		task ("Unlink") {
+			AssertElementPresent(locator1 = "CKEditorBalloonEditorSample#HYPERLINK_UNLINK");
+
+			Click(locator1 = "CKEditorBalloonEditorSample#HYPERLINK_UNLINK");
+
+			AssertElementNotPresent(locator1 = "CKEditorBalloonEditorSample#BODY_TEXT_HYPERLINK");
+		}
+
+		task ("Link") {
+			TripleClick(locator1 = "CKEditorBalloonEditorSample#BODY_TEXT");
+
+			Click(locator1 = "CKEditorBalloonEditorSample#TOOLBAR_LINK_BUTTON");
+
+			Type.sendKeys(
+				locator1 = "CKEditorBalloonEditorSample#HYPERLINK_TEXT_INPUT",
+				value1 = "www.google.com");
+
+			Click(locator1 = "CKEditorBalloonEditorSample#CHK_BTN");
+
+			AssertVisible(locator1 = "CKEditorBalloonEditorSample#BODY_TEXT_HYPERLINK");
+		}
+	}
+
+	@description = "Assert table option enabled in the toolbar"
+	@priority = "4"
+	test CanManageTable {
+		task ("Create a table") {
+			BalloonEditor.clearParagraphWithBackspace(paragraph = "CKEditorBalloonEditorSample#BODY_TEXT_HYPERLINK");
```

Comment:
Hmm...this looks like you're assuming `CanManageTable` is run right after `CanManageHyperlink`. But we should run the testcases independent of each other.

Reply:
also, paragraph shouldn't be a valid parameter to pass

Reply:
the hyperlink appears as a precondition in the sample, it is shorter that is why I have chosen to interact with it. I have already simplified it though :)